Refactor RevealItem to replace RVItem for IPC messaging

beidson opened the pull request:
#### 413e473787ebd7cd36840613a9a3d6216ab18c6b
<pre>
Refactor RevealItem to replace RVItem for IPC messaging
<a href="https://bugs.webkit.org/show_bug.cgi?id=264065">https://bugs.webkit.org/show_bug.cgi?id=264065</a>
<a href="https://rdar.apple.com/117822336">rdar://117822336</a>

Reviewed by Megan Gardner.

The only RVItems that RevealItem every serialized for IPC were created internally with text and a range.
This patch changes serializing the RVItem to simply serializing the text and range.

* Source/WebCore/editing/cocoa/DictionaryLookup.mm:
* Source/WebKit/Shared/Cocoa/RevealItem.h:
(WebKit::RevealItemRange::RevealItemRange):
(WebKit::RevealItem::text const):
(WebKit::RevealItem::selectedRange const):
(WebKit::RevealItem::item const): Deleted.
* Source/WebKit/Shared/Cocoa/RevealItem.mm:
(WebKit::RevealItemRange::RevealItemRange):
(WebKit::RevealItem::RevealItem):
(WebKit::RevealItem::highlightRange const):
(WebKit::RevealItem::item const):
* Source/WebKit/Shared/Cocoa/RevealItem.serialization.in:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::revealItemForCurrentSelection):

Canonical link: <a href="https://commits.webkit.org/270131@main">https://commits.webkit.org/270131@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6eec41475e420eb4d179a105704dfc300a206097

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24559 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3108 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25819 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26685 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22571 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24828 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4794 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/542 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22966 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24804 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2189 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21215 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27268 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1915 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22141 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28328 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22422 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22472 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26121 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1824 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/160 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3135 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5902 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2281 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2197 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

megangardner commented on Source/WebKit/Shared/Cocoa/RevealItem.h:
```diff
@@ -27,24 +27,44 @@
+    }
+
+    NSUInteger location { 0 };
+    NSUInteger length { 0 };
```
Is there a reason you're choosing to store this as the parts of an NSRange instead of an NSRange?

beidson replied:
Yes! Teaching the serializer generator to handle a raw C struct typedef like NSRange turned out to be extremely painful!

If that capability gets added to the generator in the future, we can ditch the range wrapper and just go to a raw NSRange

megangardner replied:
Maybe add a note to that effect? And/or file a radar?

megangardner reviewed (approved):
Seems reasonable! Just one question, but otherwise seems good!

beidson committed:
Refactor RevealItem to replace RVItem for IPC messaging
https://bugs.webkit.org/show_bug.cgi?id=264065
rdar://117822336

Reviewed by Megan Gardner.

The only RVItems that RevealItem every serialized for IPC were created internally with text and a range.
This patch changes serializing the RVItem to simply serializing the text and range.

* Source/WebCore/editing/cocoa/DictionaryLookup.mm:
* Source/WebKit/Shared/Cocoa/RevealItem.h:
(WebKit::RevealItemRange::RevealItemRange):
(WebKit::RevealItem::text const):
(WebKit::RevealItem::selectedRange const):
(WebKit::RevealItem::item const): Deleted.
* Source/WebKit/Shared/Cocoa/RevealItem.mm:
(WebKit::RevealItemRange::RevealItemRange):
(WebKit::RevealItem::RevealItem):
(WebKit::RevealItem::highlightRange const):
(WebKit::RevealItem::item const):
* Source/WebKit/Shared/Cocoa/RevealItem.serialization.in:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::revealItemForCurrentSelection):

Canonical link: https://commits.webkit.org/270131@main
